include/seqan3/std/concept/object_regular.hpp

```diff
@@ -0,0 +1,63 @@
+// ============================================================================
+//                 SeqAn - The Library for Sequence Analysis
+// ============================================================================
+//
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Adaptions of core language concepts.
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/std/concept/comparison.hpp>
+#include <seqan3/std/concept/object.hpp>
+
+namespace seqan3
+{
+
+/*!\addtogroup concept
+ * \{
+ */
+
+/*!\interface   seqan3::regular_concept
+ * \brief       Subsumes seqan3::semi_regular_concept and seqan3::equality_comparable_concept.
+ * \extends     seqan3::semi_regular_concept
+ * \extends     seqan3::equality_comparable_concept
+ * \sa          http://en.cppreference.com/w/cpp/experimental/ranges/concepts/Regular
+ */
+//!\cond
+template <typename t>
+concept bool regular_concept = semi_regular_concept<t> && equality_comparable_concept<t>;
```

Comment:
Please give me a really good explanation why you moved this into it's own header.

Reply:
there now is a circular dependency between object.hpp and comparison.hpp and since you can' forward declare concepts, I needed to split out one concept from either object or comparison...

Reply:
Boah, we should really only use `std/concepts/all.hpp`

Reply:
Yeah, I am thinking about making the headers in the std module be exactly like in the STL which means we would just have one header anyway.